src/utils/registerNewNetwork.ts

```diff
@@ -1,42 +1,54 @@
 import { JsonRpcProvider } from '@ethersproject/providers';
+import { Network } from '@ethersproject/networks';
 import {
   L1Network,
   L2Network,
   addCustomNetwork,
   constants as arbitrumSdkConstants,
 } from '@arbitrum/sdk';
-import {
-  getL1Network,
-  getL2Network,
-  l1Networks,
-  l2Networks,
-} from '@arbitrum/sdk/dist/lib/dataEntities/networks';
+import { getL1Network, getL2Network } from '@arbitrum/sdk/dist/lib/dataEntities/networks';
 import { RollupAdminLogic__factory } from '@arbitrum/sdk/dist/lib/abi/factories/RollupAdminLogic__factory';
+import { isArbitrumChain } from '@arbitrum/sdk/dist/lib/utils/lib';
+import { arbitrum, arbitrumNova, arbitrumSepolia, mainnet, sepolia } from 'viem/chains';
 
-export const registerNewNetwork = async (
-  l1Provider: JsonRpcProvider,
-  l2Provider: JsonRpcProvider,
-  rollupAddress: string,
-): Promise<{
-  l1Network: L1Network;
-  l2Network: Omit<L2Network, 'tokenBridge'>;
-}> => {
-  const l1NetworkInfo = await l1Provider.getNetwork();
-  const l2NetworkInfo = await l2Provider.getNetwork();
+async function isNetworkRegistered(provider: JsonRpcProvider, { type }: { type: 'L1' | 'L2' }) {
+  try {
+    if (type === 'L1') {
+      await getL1Network(provider);
+      return true;
+    } else {
+      await getL2Network(provider);
+      return true;
+    }
+  } catch (e) {
+    return false;
+  }
+}
 
-  const l1Network: L1Network = {
-    blockTime: 10,
-    chainID: l1NetworkInfo.chainId,
-    explorerUrl: '',
-    isCustom: true,
-    name: l1NetworkInfo.name,
-    partnerChainIDs: [l2NetworkInfo.chainId],
-    isArbitrum: false,
-  };
+const parentChainIdFromChildChainId = (childChainId: number) => {
+  return (
+    {
+      [arbitrum.id]: mainnet.id,
+      [arbitrumNova.id]: mainnet.id,
+      [arbitrumSepolia.id]: sepolia.id,
+    }[childChainId] || sepolia.id
+  );
```

Comment:
I think we also want the nitro testnode L2 here, and maybe not default sepolia and just let it error?